teps/0091-trusted-resources.md

```diff
@@ -230,71 +215,168 @@ spec:
     resources: {}
 ```
 
-`tekton.dev/signature` is used to store the signature.
+Signed Pipeline
+The sample signed pipeline file looks like this:
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  annotations:
+    tekton.dev/signature: MEQCIHhFC480mv7mh/6dxAlp/mvuvGXanuSghMsT+iBhWxt5AiBDvHv8sfKjJ3Ozrzvp+cjr30AOx1SPQDCcaJjlpAoVrA==
+  name: example-Pipeline
+spec:
+  tasks:
+    - name: example-task
+      taskRef:
+        name: example-task
+```
+
+`tekton.dev/signature` is used to store the signature. The future work could be to store the signature in a separate file along with the resource. Remote resolution should be updated to resolve both the resource and the signature.
 
-The signed Resource can be installed directly on the kubernetes cluster or built as OCI bundle to be stored in the registry.
+The signed Resource can be installed directly on the Kubernetes cluster or built as OCI bundle to be stored in the registry.
 
 ### Verify the Resources
 
-The verification should be done in the Tekton Pipeline's admission webhook. A new configmap should be added to gate the verification
-code.
+The verification should be done in the Tekton Pipeline's reconciler after remote resolution. A new configmap should be added to gate the verification code.
+
+Before the verification, the signature is extracted from the resource. For the public key there are several options we can choose by configuration.
+
+In this TEP we propose to store the public keys on the Kubernetes installation. For example, the keys can be configured in a new CRD and deployed to the cluster. This can help to support multiple keys, validate the spec, and have dedicated RBAC policies, so no other components in the cluster can modify the keys.
+
+Example of the Key CRD:
+```yaml
+apiVersion: tekton.dev/v1beta1
+Kind: VerificationPolicy
+metadata:
+ name: verification-policy
+ namespace: tekton-pipelines
+spec:
+   Authorities:
+    - name: cosign
+      key: /etc/signing-secrets/cosign.pub
+    - name: kms
+      key: gcpkms://projects/<project>/locations/<location>/keyRings/<keyring>/cryptoKeys/<key>
+    - name: keyless
+      keyless:
+        url: "https://fulcio.sigstore.dev"
+```
+
+API (Reference from [policy-controller](https://github.com/sigstore/policy-controller)):
+```go
+type VerificationPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	// Spec holds the desired state of the VerificationPolicy.
+	Spec VerificationPolicySpec `json:"spec"`
+}
+
+type VerificationPolicySpec struct {
+	// Resources defines the patterns of Resources names that should be subject to this policy. For example, we may want to apply this Policy only from a certain github repo. Then the ResourcesPattern should include the path.
```

Comment:
This could use a bit more details - is this [the CRD resource itself](https://github.com/tektoncd/pipeline/blob/2704d7fb38679d33f2e186a4f9b630482d53c79d/config/300-taskrun.yaml#L15-L18), an [instance of a CRD](https://github.com/tektoncd/pipeline/blob/main/examples/v1beta1/taskruns/task-result.yaml), or something else?

Examples would help here 🙏

Reply:
https://github.com/tektoncd/resolution/tree/main/gitresolver#parameters
https://github.com/tektoncd/resolution/tree/main/bundleresolver#parameters
https://github.com/tektoncd/resolution/tree/main/hubresolver#parameters
I want to use the `ResourcesPattern` to enable users to only verify resources from certain source or with certain name. 
For example, the `catalog` for hubresolver, `bundle` for bundleresolver, and use regex to match these fields.

Reply:
I'm thinking about a case where User A has a catalog repo, and he somehow exposes his key for that repo. Then we can configure the key for all resources coming from that repo